src/serialize.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+
+"use strict";
+
+import utils = require("./utils");
+import kernel = require("./kernel");
+
+import IKernelMsg = kernel.IKernelMsg;
+
+
+function _deserializeArrayBuffer(buf: ArrayBuffer): IKernelMsg {
+  var data = new DataView(buf);
+  // read the header: 1 + nbufs 32b integers
+  var nbufs = data.getUint32(0);
+  var offsets: number[] = [];
+  var i: number;
+  for (i = 1; i <= nbufs; i++) {
+    offsets.push(data.getUint32(i * 4));
+  }
+  var json_bytes = new Uint8Array(buf.slice(offsets[0], offsets[1]));
+  var msg = JSON.parse(
+    (new TextDecoder('utf8')).decode(json_bytes)
+    );
+  // the remaining chunks are stored as DataViews in msg.buffers
+  msg.buffers = [];
+  var start: number, stop: number;
+  for (i = 1; i < nbufs; i++) {
```

Comment:
`for (var i = 1, ....)`

Yes, `i` is declared above, but in the future (typescript 1.5), we'll start using `let` instead of `var`.